### PR TITLE
Data portal migration

### PR DESCRIPTION
**Pull request by mahimoksha**
We have made changes to the existing usecases to download from new data portal. We have also modified upload and download functions to use the new portal. We have changed the documentations accordingly.

### CODE REVIEW

**Comment by akenmorris** [comment]
If the user did not have an account on the old portal, the old RunUseCase script would output:

```
Step 1. Extract Data

Can't find Data/femur.zip
 _____    ___      .
|     |  /   \    / \
| ShapeWorks Portal  \
|_____|  \___/  /_____\

Downloading the femur dataset from the ShapeWorks Portal
Login info not found in current directory.
New ShapeWorks Portal users: Register an account at https://girder.shapeworks-cloud.org/#?dialog=register

Returning ShapeWorks Portal users: Enter your username and password:
```

The new one does not tell users where they can create a login:

```

Step 1. Extract Data

 _____    ___      .
|     |  /   \    / \
| ShapeWorks Portal  \
|_____|  \___/  /_____\

Getting login

Username: 
```

**Comment by akenmorris** [comment]
Should we remove the DatasetUtils python package? Is it being used anywhere?

**Comment by akenmorris** [comment]
Are we using both data portals for now?  Running the use cases, I'm getting errors about the certificate being expired for the old girder instance.  We will get this fixed but I'm wondering about why we are relying on the old one:

```
Traceback (most recent call last):
  File "/home/amorris/sci/shapeworks/Examples/Python/RunUseCase.py", line 89, in <module>
    module.Run_Pipeline(args)
  File "/home/amorris/sci/shapeworks/Examples/Python/femur_pvalues.py", line 13, in Run_Pipeline
    sw.data.download_and_unzip_dataset(dataset_name, output_directory)
  File "/home/amorris/sci/shapeworks/Python/shapeworks/shapeworks/data.py", line 102, in download_and_unzip_dataset
    DatasetUtils.downloadDataset(datasetName, destinationPath='./Data/')
  File "/home/amorris/sci/shapeworks/Python/DatasetUtilsPackage/DatasetUtils/__init__.py", line 36, in downloadDataset
    accessToken = GirderConnector.login(loginState)
  File "/home/amorris/sci/shapeworks/Python/DatasetUtilsPackage/DatasetUtils/GirderConnector.py", line 29, in login
    loginState, accessToken = _loginAndGetAccessToken()
  File "/home/amorris/sci/shapeworks/Python/DatasetUtilsPackage/DatasetUtils/GirderConnector.py", line 47, in _loginAndGetAccessToken
    loginState = _promptLogin()
  File "/home/amorris/sci/shapeworks/Python/DatasetUtilsPackage/DatasetUtils/GirderConnector.py", line 72, in _promptLogin
    basicAuthToken = GirderAPI.authenticateBasicAuth(serverAddress, usernamePasswordHash)
  File "/home/amorris/sci/shapeworks/Python/DatasetUtilsPackage/DatasetUtils/GirderAPI.py", line 70, in authenticateBasicAuth
    response = _makeGetRequest(
  File "/home/amorris/sci/shapeworks/Python/DatasetUtilsPackage/DatasetUtils/GirderAPI.py", line 51, in _makeGetRequest
    return _makeRequest(requestFunction = requests.get, url = url, params = params, headers = headers, actionMessage = actionMessage, data = None, printError = printError)
  File "/home/amorris/sci/shapeworks/Python/DatasetUtilsPackage/DatasetUtils/GirderAPI.py", line 29, in _makeRequest
    response = requestFunction(url = url, params = params, headers = headers, stream = True)
  File "/home/amorris/miniconda3/envs/shapeworks/lib/python3.9/site-packages/requests/api.py", line 75, in get
    return request('get', url, params=params, **kwargs)
  File "/home/amorris/miniconda3/envs/shapeworks/lib/python3.9/site-packages/requests/api.py", line 61, in request
    return session.request(method=method, url=url, **kwargs)
  File "/home/amorris/miniconda3/envs/shapeworks/lib/python3.9/site-packages/requests/sessions.py", line 529, in request
    resp = self.send(prep, **send_kwargs)
  File "/home/amorris/miniconda3/envs/shapeworks/lib/python3.9/site-packages/requests/sessions.py", line 645, in send
    r = adapter.send(request, **kwargs)
  File "/home/amorris/miniconda3/envs/shapeworks/lib/python3.9/site-packages/requests/adapters.py", line 517, in send
    raise SSLError(e, request=request)
requests.exceptions.SSLError: HTTPSConnectionPool(host='girder.shapeworks-cloud.org', port=443): Max retries exceeded with url: /api/v1/user/authentication (Caused by SSLError(SSLCertVerificationError(1, '[SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: certificate has expired (_ssl.c:1129)')))
/home/amorris/sci/shapeworks/Testing/UseCaseTests/UseCaseTests.cpp:31: Failure
Value of: result
```

**Comment by akenmorris** [comment]
Do the `UseCaseTests` work for you?  I have the following failures:

[  FAILED  ] 5 tests, listed below:
[  FAILED  ] UseCaseTests.ellipsoid_fd
[  FAILED  ] UseCaseTests.ellipsoid_evaluate
[  FAILED  ] UseCaseTests.deep_ssm
[  FAILED  ] UseCaseTests.ellipsoid_pca
[  FAILED  ] UseCaseTests.femur_pvalues

To get the GitHub actions runners to use the new portal, I'll have to create an account for it and set up the environment variable as we did for the old one.

File: Python/shapeworks/shapeworks/upload.py [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+            description='Project created via SWCC',
+            dataset=dataset,
+        ).create()
+        print(project)
+
+        print('Done. \n')
+
+if __name__=="__main__":
+    dataset_names = ["femur_pvalues"]
+    project_files = ["/home/sci/mkaranam/Desktop/ShapeWorksLatest/ShapeWorks/Examples/Python/Output/femur_pvalues/femur_pvalues/femur_pvalues.swproj"]
```
Comment (iyerkrithika21) [inline Python/shapeworks/shapeworks/upload.py:70]: This file has to be cleaned up. We can also add comments to show how to upload but let's remove the main function from this file.

File: Python/shapeworks/shapeworks/portal.py [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+            if not download_path.exists():
+                rmtree(str(download_path))
+            # project = Project()   
+            for project in dataset.projects:
+                project.download(Path(download_path))
+                break
+            # project = Project.from_id(dataset.id)
+            # project.download(download_path)
+
+    # with swcc_session() as session:
```
Comment (iyerkrithika21) [inline Python/shapeworks/shapeworks/portal.py:180]: Clean up file

**Review by iyerkrithika21 — CHANGES_REQUESTED** [review]
For the all the use cases, code has to be cleaned up to remove the commented-out old code

**Comment by akenmorris** [comment]
Running the VerifyUseCases script on this branch, I got the following failures:

```
Comparing compactness: 0.73765 vs 0.73938: ok
Comparing generalization: 3.11647 vs 2.59637: different (0.520094)
Comparing specificity: 1.78638 vs 1.51886: ok
Error: particle system did not match ground truth
left_atrium --verify :    632 seconds : FAILED
```

```
Comparing compactness: 0.998934 vs 0.994156: ok
Comparing generalization: 0.184094 vs 0.332373: different (0.148279)
Comparing specificity: 0.491375 vs 0.406605: ok
Error: particle system did not match ground truth
ellipsoid_mesh --verify :     15 seconds : FAILED
```

Are we sure it's using the same data?

**Comment by akenmorris** [comment]
It seems that if I run `python RunUseCase.py ellipsoid_mesh --verify --tiny_test` before `python RunUseCase.py ellipsoid_mesh --verify`, then it uses the tiny test data for the full test and then fails.

**Comment by zahidemon** [comment]
The verification error was because of a bug in the existing dataset check. I fixed it and ran the  VerifyUseCases script again. Every usecase passed

**Pull request merged**